client/src/components/SystemStatus/SystemStatus.jsx

```diff
@@ -0,0 +1,141 @@
+  var sidebarItems = [];
+
+  systemList.forEach((system) => {
+    sidebarItems.push({
+      to: `${root}/${system.hostname}`,
+      label: `${system.display_name}`,
+      disabled: false,
+      hidden: false,
+      iconName: '',
+    });
+  });
+
+  return <Sidebar sidebarItems={sidebarItems} />;
+};
+
+const Layout = ({ hostname }) => {
+  const {
+    loading,
+    error: loadingError,
+    list: systemList,
+  } = useSelector((state) => state.systemMonitor);
+
+  const selectedSystem = systemList.find((sys) => sys.hostname === hostname);
+
+  if (loadingError) {
+    return (
+      <Section className={styles['no-results-message']}>
+        <SectionMessage type="info">
+          Unable to gather system information
+        </SectionMessage>
+      </Section>
+    );
+  }
+
+  return (
+    <>
+      <Section
+        className={styles['root']}
+        contentLayoutName="twoColumn"
+        contentClassName={styles['layout']}
+        content={
+          <div className={styles['panel-1']}>
+            <Sysmon system={selectedSystem?.hostname} />
+            {loading ? (
+              <LoadingSpinner />
+            ) : (
+              <SectionTableWrapper
+                className={styles['content']}
+                contentShouldScroll
+              >
+                <SystemStatusQueueTable system={selectedSystem} />
+              </SectionTableWrapper>
+            )}
+          </div>
+        }
+        /* For Average Wait Times table, add a second <div> */
+      />
+    </>
+  );
+};
+
+const DefaultSystemRedirect = () => {
+  const systemList = useSelector((state) => state.systemMonitor.list);
+
+  const history = useHistory();
+  useEffect(() => {
+    if (systemList.length === 0) return;
+    const defaultSystem =
+      systemList.find((sys) => sys.display_name === 'Frontera') ??
+      systemList[0];
+    history.push(`/workbench/system-status/${defaultSystem.hostname}`);
+  }, [systemList]);
+  return <></>;
+};
+
+const Routes = () => {
+  return (
+    <Switch>
+      <Route
+        path={`${root}/:hostname`}
+        render={({ match: { params } }) => {
+          return <Layout hostname={params.hostname} />;
+        }}
+      />
+      <Route path={`${root}`}>
+        <DefaultSystemRedirect />
+      </Route>
+    </Switch>
+  );
+};
+
+const SystemStatus = () => {
+  const { list: systemList } = useSelector((state) => state.systemMonitor);
+
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    if (systemList.length == 0) {
+      dispatch({ type: 'GET_SYSTEM_MONITOR' });
+    }
+  }, [dispatch]);
+
+  const location = useLocation();
+
+  const hostname = location.pathname.split('/')[3];
+
+  const selectedSystem = systemList.find((sys) => sys.hostname === hostname);
+
+  return (
+    <Section
+      bodyClassName="has-loaded-system-status"
+      messageComponentName="SYSTEM STATUS"
+      header={`System Status / ${selectedSystem?.display_name}`}
```

Comment:
If you navigate to a status url directly, e.g. https://cep.test/workbench/system-status/frontera.tacc.utexas.edu, for a brief moment it displays `"System Status / undefined"`
<img width="836" alt="Screenshot 2023-06-30 at 10 54 57 AM" src="https://github.com/TACC/Core-Portal/assets/20326896/e08c98e7-4009-48bd-bd24-e09d96b06045">

Reply:
Also, when the error finally appears, it is very tall. Other sections fix this with similar one-off CSS rules. I do not know how. Feel free to mimic their solution.[^1]

[^1]: What?! Who are you and what have you done with @wesleyboar? I am still here, and me. Yes, there should be one way to do this. I am still on TUP work, so I can't jump on every Portal CSS improvement opportunity.

Reply:
![tall error](https://github.com/TACC/Core-Portal/assets/62723358/3ae1b60d-f96e-4eac-93b8-84f76985e6ac)